src/bridge-deployer/src/canister_ids/canisters.rs

```diff
@@ -0,0 +1,92 @@
+use std::fmt;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+const BRC20_BRIDGE_NAME: &str = "brc20-bridge";
+const BTC_BRIDGE_NAME: &str = "btc-bridge";
+const ERC20_BRIDGE_NAME: &str = "erc20-bridge";
+const ICRC2_BRIDGE_NAME: &str = "icrc2-bridge";
+const RUNE_BRIDGE_NAME: &str = "rune-bridge";
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum Canister {
+    Brc20,
+    Btc,
+    Erc20,
+    Icrc2,
+    Rune,
+}
+
+impl fmt::Display for Canister {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let canister = match self {
+            Canister::Brc20 => BRC20_BRIDGE_NAME,
+            Canister::Btc => BTC_BRIDGE_NAME,
+            Canister::Erc20 => ERC20_BRIDGE_NAME,
+            Canister::Icrc2 => ICRC2_BRIDGE_NAME,
+            Canister::Rune => RUNE_BRIDGE_NAME,
+        };
+
+        write!(f, "{}", canister)
+    }
+}
+
+impl FromStr for Canister {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            BRC20_BRIDGE_NAME => Ok(Self::Brc20),
+            BTC_BRIDGE_NAME => Ok(Self::Btc),
+            ERC20_BRIDGE_NAME => Ok(Self::Erc20),
+            ICRC2_BRIDGE_NAME => Ok(Self::Icrc2),
+            RUNE_BRIDGE_NAME => Ok(Self::Rune),
+            _ => Err("invalid canister"),
+        }
+    }
+}
+
+impl Serialize for Canister {
```

Comment:
Why do we need custom serialization? In any case, add (de)serialization tests

Reply:
Because otherwise it won't use the impl of ToString and FromStr we've implemented